src/core/a-scene.js

```diff
@@ -402,13 +402,25 @@ var AScene = module.exports = registerElement('a-scene', {
 
     setupCanvas: {
       value: function () {
-        var canvas = this.canvas = document.createElement('canvas');
+        var canvasSelector = this.getAttribute('canvas');
+        var canvas;
+
+        if (canvasSelector) {
+          canvas = this.canvas = document.querySelector(canvasSelector);
+        } else {
+          canvas = this.canvas = document.createElement('canvas');
+        }
+
         canvas.classList.add('a-canvas');
         // Prevents overscroll on mobile devices.
         canvas.addEventListener('touchmove', function (evt) {
           evt.preventDefault();
         });
-        document.body.appendChild(canvas);
+
+        if (!canvasSelector) {
```

Comment:
You can also move this statement to the appropriate branch of the if above https://github.com/aframevr/aframe-core/pull/763/files#diff-6bda77095e5056661a0b803e70987f87R408